src/error.rs

```diff
@@ -0,0 +1,16 @@
+#[derive(Debug)]
+pub enum Error {
+    CargoAdd(String),
+    ChangeDir,
+    CreateCargo,
+    CreateFile(&'static str),
+    CreateFolder(&'static str),
```

Comment:
I appreciate the effort in the interest of performance, but I don't like the inconsistency and rigidness of using a static string slice instead of an allocated string. What if `CreateFile` or `CreateFolder` need to be invoked in a context similar to `CargoAdd` where the name is not known until runtime?

Reply:
Yeah I probably am a little too rigid myself regarding allocation, especially for error cases.
Reverted back to `String`s.